File: 1437-check-if-all-1s-are-at-least-length-k-places-away/1437-check-if-all-1s-are-at-least-length-k-places-away.js
var kLengthApart = function (nums, k) {
    let lastIdx = -1;

    for (let i = 0; i < nums.length; i++) {
        if (nums[i] === 1) {
            if (lastIdx !== -1 && i - lastIdx - 1 < k) {
                return false;
            }
            lastIdx = i;
        }
    }

    return true;
};

File: 1437-check-if-all-1s-are-at-least-length-k-places-away/1437-check-if-all-1s-are-at-least-length-k-places-away.cs
public class Solution {
    public bool KLengthApart(int[] nums, int k) {
        int Idx = -1;
        for (int i = 0; i < nums.Length; i++)
        {
            if (nums[i] == 1)
            {
                if (Idx != -1 && i - Idx - 1 < k) return false;
                Idx = i;
            }
        }
        return true;
    }
}